src/core/evaluator.js

```diff
@@ -4420,7 +4417,13 @@ class PartialEvaluator {
 
     const fontNameStr = fontName?.name;
     const baseFontStr = baseFont?.name;
-    if (!isType3Font && fontNameStr !== baseFontStr) {
+    if (isType3Font) {
+      if (!fontNameStr) {
+        // Since the /FontDescriptor is optional in Type3 fonts, ensure that we
+        // always have a "valid" /FontName (fixes issue19954.pdf).
+        fontName = Name.get(type);
+      }
+    } else if (fontNameStr !== baseFontStr) {
```

Comment:
There is a `fontName ||= baseFont` further down that can be pulled inside this `else if`. It's only useful when `fontName` is missing and `baseFont` is not, which means that we do not have a type 3 font and that `fontNameStr !== baseFontStr`.